.github/workflows/workflow.yml

```
@@ -1,60 +1,112 @@
 name: Test and Deploy
 on:
   pull_request:
-    branches: 
-      - main
-  push:
     branches:
       - main
+  push:
   create:
     tags:
       - '*'
 jobs:
   check:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
       with:
-        python-version: 3.8
+        python-version: '3.x'
+        cache: pip
+        cache-dependency-path: .github/workflows/workflow.yml
     - name: Black
       run: |
         pip install black
         black --check --exclude /docs --diff .
 
-  test-extension:
+  build-wheel:
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.7"
+          cache: pip
+          cache-dependency-path: |
+            .github/workflows/workflow.yml
+            dev-requirements.txt
+      - name: Install dependencies
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install -r dev-requirements.txt
+      - name: Install package
+        run: |
+          python -m pip install .
+      - name: Build wheel
+        run: |
+          python -m pip install build
+          python -m build
+      - name: Upload wheel as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: my-wheel
+          path: dist/*.whl
+
+  test-extension:
```

```suggestion
  test:
```
The "Checks" sidebar has a character limit before cutting off.